pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -128,22 +126,9 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 			hyd.Labels[constant.InfraLabelName] = hyd.Spec.InfraID
 		}
 
-		if err := r.patchHypershiftDeploymentResource(&hyd, &oHyd); err != nil || hyd.Spec.InfraID == "" {
+		if err := r.patchHypershiftDeploymentResource(&hyd); err != nil || hyd.Spec.InfraID == "" {
 			return ctrl.Result{}, fmt.Errorf("failed to update infra-id: \"%s\" and error: %w", hyd.Spec.InfraID, err)
 		}
-
```

Comment:
This code was removed from the finalizer section, and placed in the individual INFRA create sections, so that problem conditions would be replaces with configuring conditions when the controller re-reconciles.